lib/mayaUsd/render/mayaToHydra/renderOverride.cpp

```diff
@@ -533,6 +537,13 @@ MStatus MtohRenderOverride::Render(const MHWRender::MDrawContext& drawContext)
 
     if (!_initializedViewport) {
         _InitHydraResources();
+
+        if (!_taskController) {
+            // No task controller means something has gone wrong, use this as a signal so we
+            // aren't endlessly failing to initialize.
```

Comment:
As you say, we're essentially using `_taskController` as a indicator for whether initialization succeeded; however, I think it would be better if we had a new dedicated member variable, say `_initializationSucceeded`, for this purpose - both for clearer naming, and because there's a decent chunk of stuff done AFTER the `_taskController` is created.  We could then also rename `_initializatedViewport` to `_initializationAttempted` - since that's effectively how it's being used now - and unconditionally set it as soon as `_InitHydraResources()` starts.  Again, this is effectively how it's already functioning, since it's set to true whether it succeeds or fails, but it's obscured by the fact that it's set to `true` in two different places.

Between the naming and placement of where these are set (at the very start and end of `_InitHydraResources()`), I think the intent will be a lot clearer, without the need for explanation.